deps/rabbit_common/src/rabbit_writer.erl

```diff
@@ -265,10 +265,6 @@ handle_message({send_command_and_notify, QPid, ChPid, MethodRecord, Content},
 handle_message({'DOWN', _MRef, process, QPid, _Reason}, State) ->
     rabbit_amqqueue_common:notify_sent_queue_down(QPid),
     State;
-handle_message({inet_reply, _, ok}, State) ->
-    rabbit_event:ensure_stats_timer(State, #wstate.stats_timer, emit_stats);
```

Comment:
Very good spot, thank you!
Yes, it seems that the AMQP writers deviate from what's described in https://github.com/rabbitmq/rabbitmq-server/blob/b959a92cfa6961e0bc98986e3ca0bee45ae0945a/deps/rabbit_common/src/rabbit_event.erl#L68-L90 in the sense that they don't stop the stats timer before hibernation (and don't start the stats timer upon wakeup). Since the "connection" stats emitting object is shared between two processes (reader & writer), the writer just let's the reader know to emit stats when the writer's timer fires. Furthermore the writer needs to ensure_stats_timer after having written to the socket.
So, I added that behaviour back.